File: src/50-59/58.py
RUN = 10000
TARGET  = 0.1

i = 1
m = 1
n = 1

prime_count = 0
total_count = 0

answer = -1

def isprime(n):
    return n>1 and all(n%i for i in range(2,int(n**0.5)+1))

x = -1
while True:
    x+=1
    for y in range(4):
        i=i+n
        total_count += 1
        if y == 0:
            prime_count += 1 if isprime(i-1) else 0
            # print(prime_count/total_count,x)
            if prime_count and prime_count/total_count<TARGET:
                print(x*2+1)
                exit()
                
        else:
            prime_count += 1 if isprime(i) else 0
        
        m,n=(m-1,n)if m else(1,n+1)

File: src/50-59/58.c
#include <stdio.h>
#include <stdbool.h>

#define RUN 10000
#define TARGET 0.1

bool isprime(long n) {
    if (n <= 1)
        return false;
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main() {
    long i = 1;
    long m = 1;
    long n = 1;

    int prime_count = 0;
    int total_count = 0;

    int x = -1;
    while (1) {
        x++;
        for (int y = 0; y < 4; y++) {
            i += n;

            total_count++;
            if (y == 0) {
                prime_count += isprime(i - 1) ? 1 : 0;
                if (prime_count && (double)prime_count / total_count < TARGET) {
                    printf("%d\n", x * 2 + 1);
                    return 0;
                }
            } else {
                prime_count += isprime(i) ? 1 : 0;
            }
            if (m) {
                m--;
            } else {
                m = 1;
                n = n + 1;
            }
        }
    }
    return 0;
}
